src/peakrdl_sv/regmodel.py

```diff
@@ -22,8 +52,357 @@ def _parse_rdl(self, rdlfile: str) -> AddressMap:
         listener.walk(root, unroll=True)
         return listener.top_node
 
-    def get_register_by_name(self, name: str) -> Register:
-        for register in self.top_node.get_registers():
-            if register.name == name:
-                return register
-        return None
+    def _map_registers(self) -> dict[str, Register]:
+        """Builds a dictionary which maps register relative paths to the register
+        itself, from which its properties may be extracted
+
+        :return: A map of register relative paths to Register objects
+        :rtype: Dict[str, Register]
+        """
+        return {
+            register.path("."): register for register in self._top_node.get_registers()
+        }
+
+    def _map_fields(self) -> dict[str, dict[str, FieldWrapper]]:
+        """Maps register names to dictionaries of fields which maps to (desired) values
+
+        :raises AttributeError: _map_fields called before _map_registers
+        :return: A dict of {register names : dict of {field names : (field, value)}}
+        :rtype: Dict[str, Dict[str, FieldWrapper]]
+        """
+
+        if not hasattr(self, "_reg_map"):
```

Comment:
You're raising an exception here on a private method that won't be called by anyone/anything other than the constructor.  You could merge these mapping methods into a single one or have a parent that calls them.  A simple `assert self._reg_map is not None` might also be cleaner.  I think raising an `AttributeError` is probably a better error when an incorrect attribute is passed in to a method.

Reply:
If `self._reg_map` has not been created (i.e. `self._map_registers()` has not been called first) then performing an `assert self._reg_map is not None` will raise an attribute error as `self._reg_map` doesnt exist. With that said, I agree the use of `AttributeError` is probably not a good idea here, I think a better approach is a `try...excecpt` on the return statement which catches the `AttributeError` if there is incorrect access to `self._reg_map` and re-raises the exception.